WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentsAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package org.wordpress.android.ui.sitecreation.segments
+
+import android.support.v7.util.DiffUtil
+import android.support.v7.widget.RecyclerView.Adapter
+import android.view.ViewGroup
+import org.wordpress.android.fluxc.model.vertical.VerticalSegmentModel
+import org.wordpress.android.ui.sitecreation.segments.NewSiteCreationSegmentViewHolder.SegmentViewHolder
+import org.wordpress.android.util.image.ImageManager
+
+class NewSiteCreationSegmentsAdapter(
+    private val onItemTapped: (VerticalSegmentModel) -> Unit = { },
```

Comment:
This is actually a good case for why we should not use the `VerticalModel` directly and have an adapter item. For the item tapped action, the adapter plays the role of a middle man where it doesn't contribute to anything. If we had the adapter item, we could have separate `onItemTapped` actions for each adapter item and the Adapter or the ViewHolder wouldn't actually need to know anything about the `VerticalModel`.

Again, I am not actually asking you to make this change. You mentioned you were curious about PR reviews and I just wanted to be more elaborate on my thoughts when I am reviewing this PR in case you find it useful. I normally don't bring up all these things, so if you don't find it useful, I am happy to ignore the things we don't need yet.

Reply:
I'll be happy to make this change and I enjoy these comments as they make me think + I learn from them.
I'll wait for you comment https://github.com/wordpress-mobile/WordPress-Android/pull/8547#discussion_r231315975 before I make the change, so I'm sure we are on the same page.